src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,284 @@
+                throw new DukeException("This task does not exist. Please provide a valid task index.");
+            } else {
+                printLine();
+                Task deletedTask = tasks.get(taskNumber - 1);
+                tasks.remove(taskNumber - 1);
+                System.out.println("Noted. I've removed this task:");
+                System.out.print("\t  ");
+                System.out.println(deletedTask);
+                System.out.println("Now you have " + tasks.size() + " tasks in your list.");
+                printLine();
+                saveTask();
+            }
+        }
+
+        private static void printList () throws DukeException {
+            if (tasks.size() == 0) {
+                throw new DukeException("Sorry, there are no tasks in the list currently.");
+            } else {
+                printLine();
+                System.out.println("Here are the tasks in your list:");
+                for (int i = 0; i < tasks.size(); i++) {
+                    System.out.print("\t" + (i + 1) + ".");
+                    System.out.println(tasks.get(i));
+                }
+            }
+            printLine();
+        }
+
+        private static void addTodo (String input) throws DukeException, IOException {
+            String[] words = input.split(" ");
+            if (words.length < 2) {
+                throw new DukeException("☹ OOPS!!! The description of a todo cannot be empty.");
+            } else {
+                Todo todo = new Todo(input.substring(5));
+                tasks.add(todo);
+                printLine();
+                System.out.println("Got it. I've added this task:");
+                System.out.print("\t  ");
+                System.out.println(todo);
+                System.out.println("Now you have " + tasks.size() + " tasks in the list.");
+                printLine();
+                saveTask();
+            }
+        }
+
+        private static void addDeadline (String input) throws DukeException, IOException {
+            String[] words = input.split(" ");
+            int index = input.indexOf("/");
+            if (words.length < 2 || index == -1) {
+                throw new DukeException("☹ OOPS!!! The description or date of a deadline cannot be empty.");
+            } else {
+                String[] wordDeadline = input.substring(9).split("/");
+                Deadline deadline = new Deadline(wordDeadline[0].trim(), wordDeadline[1].substring(3));
+                tasks.add(deadline);
+                printLine();
+                System.out.println("Got it. I've added this task:");
+                System.out.print("\t  ");
+                System.out.println(deadline);
+                System.out.println("Now you have " + tasks.size() + " tasks in the list.");
+                printLine();
+                saveTask();
+            }
+        }
+
+        private static void addEvent (String input) throws DukeException, IOException {
```

Comment:
There should be no space between method name and parameter. Same with other methods